pyodide_build/recipe/graph_builder.py

```diff
@@ -89,6 +89,9 @@ def __init__(self, pkgdir: Path, config: MetaConfig):
         self.unbuilt_host_dependencies = set(self.host_dependencies)
         self.host_dependents = set()
 
+    def __hash__(self) -> int:
+        return hash((self.name, self.version))
```

Comment:
I am curious why pre-commit wants this field

Reply:
Oh, it's for https://docs.astral.sh/ruff/rules/eq-without-hash/, so if we have `__eq__`, we must have `__hash__` too. I used a tuple with a name and a version, because it is the minimally constructible case where we will have a unique hash – no two recipes can have the same name and version.

I'll open a follow-up PR with a more semantic change that should clear the logic up here.